src/include/main/jucc.h

```diff
@@ -1,15 +1,41 @@
 #ifndef JUCC_JUCC_H
 #define JUCC_JUCC_H
 
+#include <algorithm>
+#include <fstream>
 #include <iostream>
 #include <string>
+#include <vector>
+
+#include "grammar/grammar.h"
+#include "lexer/lexer.h"
+#include "parsing/parsing.h"
+#include "parsing/utils/parsing_table.h"
+#include "utils/first_follow.h"
+#include "utils/left_factoring.h"
+#include "utils/left_recursion.h"
 
 namespace jucc {
 /**
  * Dummy function to test setup clang and build config
  */
 std::string Hello();
 
+class InputParser {
+  std::vector<std::string> tokens_;
+
+ public:
+  InputParser(int argc, char *argv[]) {
```

Comment:
Add comments to header file.
Move function body to cpp file.
Only one line function bodies/ constructors are allowed.

Reply:
Will look into this  👍